### DISCUSSION TITLE
Classic Wooden Twister Roller Coaster

### DISCUSSION THREAD

**Discussion by ZeeMaji** [OpenRCT2/OpenRCT2#21502] [Feature requests] [answered]
The differences between RCT1's wooden twister coaster and the RCT2 wooden coaster are pretty subtle, but to achieve max RCT1 parity it should be implemented eventually.

This seems like a good canidate for a ride type & track type object once those are possible to make as it seems quite painful to do with the currently hard coded tracks.

The main differences on the wooden twister are:
1. The track has a different banking angle that's mildly shallow, it's between the RCT1 classic wooden coaster & RCT2 wooden coaster's banking angle.
2. The trains are based upon fred church's bobs trains unlike the RCT2 wooden coaster's articulated trains which are based on millennium flyer trains. (The design of the wooden twister trains themselves suggest this, and scenarios like haunted harbor refer to it as a very old coaster type.)
3. Lower max build height. (same as classic wooden coaster)
4. No vertical loop piece.
5. No reverse incline shuttle operating mode.
6. This is technically a part of the vehicle but the ratings are slightly different as they don't have a ratingMultipler.
7. This would technically be an OpenRCT2 only difference, but the classic wooden twister would have a lower max lift speed as well (same as classic wooden coaster)

Here's the banking difference between all three tracks for comparison, order from left to right is Classic Wooden, Classic Twister, & RCT2 Wooden.
![wooden_banking_differences](https://github.com/OpenRCT2/OpenRCT2/assets/42477864/62f3fa80-c97a-49ca-ae62-35806bc2618b)

**Comment by SpartanFrederic104** [comment]
As for the description in the build window, would this one work?
_“An older-style wooden roller coaster with a fast and rough ride featuring articulated trains, plenty of air-time and twisting track”_

Does the RCT1 Wooden Twister Roller Coaster also have different track color presets of its own?

**Reply by ZeeMaji** [reply]
The description seems fine enough to me, i'm not sure if i like the "out of control feeling" bit as it seems a little repetitive with the other two wooden coaster types which also mention that but that's a pretty minor nitpick.

As for the colors, the color presets seem to be the same between the wooden twister & rct2 wooden coaster. (Excluding presets added by openrct2 of course)

**Comment by Gymnasiast** [comment]
I am okay with implementing this, seeing as we recently took a more rigorous approach to RCT1. I would, however, want to 1) fix the helix supports on the existing Wooden RC, since we’ll use its code as a starting point, and 2) do some cleanup to the drawing code of wooden roller coasters and their supports, so any new code is clean.

I think it’s fine to do within the current system, as the drawing code can simply implement just the bits that are different and relegate all other pieces to the common Wooden RC drawing code.

**Comment by Gymnasiast** [comment, marked as answer]
Has been implemented.
